tests/manage/pv_services/test_ceph_capacity_recovery.py

```diff
@@ -0,0 +1,269 @@
+
+    """
+    log.info(f"{msg}")
+    used_capacity = get_percent_used_capacity()
+    log.info(f"Used Capacity is {used_capacity}%")
+    return used_capacity
+
+
+def verify_osd_used_capacity_greater_than_expected(expected_used_capacity):
+    """
+    Verify OSD percent used capacity greate than ceph_full_ratio
+
+    Args:
+        expected_used_capacity (float): expected used capacity
+
+    Returns:
+         bool: True if used_capacity greater than expected_used_capacity, False otherwise
+
+    """
+    osds_utilization = get_osd_utilization()
+    log.info(f"osd utilization: {osds_utilization}")
+    for osd_id, osd_utilization in osds_utilization.items():
+        if osd_utilization > expected_used_capacity:
+            log.info(f"OSD ID:{osd_id}:{osd_utilization} greater than 85%")
+            return True
+    return False
+
+
+class TestCephCapacityRecovery(PASTest):
+    def setup(self):
+        """
+        Setting up test parameters
+        """
+        log.info("Starting the test setup")
+        # Run the test in its own project (namespace)
+        self.create_test_project()
+
+        self.interface = "CephFileSystem"
+
+        super(TestCephCapacityRecovery, self).setup()
+
+        # Getting the total Storage capacity
+        try:
+            self.ceph_capacity = int(self.ceph_cluster.get_ceph_capacity())
+        except Exception as err:
+            err_msg = f"Failed to get Storage capacity : {err}"
+            log.error(err_msg)
+            raise Exception(err_msg)
+
+        log.info(
+            f"Working on cluster {self.ceph_cluster.cluster_name} with capacity {self.ceph_capacity}"
+        )
+
+    def teardown(self):
+        """
+        Cleanup the test environment
+        """
+        log.info("Starting the test cleanup")
+
+        self.benchmark_obj.cleanup()
+        # Deleting the namespace used by the test
+        self.delete_test_project()
+
+        super(TestCephCapacityRecovery, self).teardown()
+
+    def test_capacity_recovery(
+        self,
+    ):
+        log.info("Ceph Recovery test start")
+        get_used_capacity("Before pulling perf image")
+
+        helpers.pull_images(constants.PERF_IMAGE)
+
+        worker_nodes_list = node.get_worker_nodes()
+        assert len(worker_nodes_list) > 1
+        node_one = worker_nodes_list[0]
+
+        self.sc_obj = ocs.OCS(
+            kind="StorageCluster",
+            metadata={
+                "namespace": self.namespace,
+                "name": constants.CEPHFILESYSTEM_SC,
+            },
+        )
+
+        used_now = get_used_capacity("After pulling perf image")
+
+        self.num_of_pvcs = 10
+        self.pvc_size = (
+            self.ceph_capacity * (1 - used_now / 100 - 0.15) / self.num_of_pvcs
+        )
+        self.pvc_size_str = str(self.pvc_size) + "Gi"
+        log.info(f"Creating pvs of {self.pvc_size_str} size")
+
+        pvc_list = []
+        pod_list = []
+        for i in range(5):
```

Comment:
dose this  need to be `range(self.num_of_pvcs)` "

Reply:
Yes, actually self.num_of_pvcs/2 , since on each loop cycel 1 pvc and 1 clone is created